src/schema/posts.ts

```diff
@@ -203,6 +203,11 @@ export const typeDefs = /* GraphQL */ `
     Whether the post's source is private or not
     """
     private: Boolean
+
+    """
+    The unix timestamp (seconds) the post will be promoted to public to
+    """
+    promoteToPublic: Int
```

Comment:
Do we want to transform it to Date before sending back to the client? Also not sure we even want to provide it to public. So maybe we should remove it even

Reply:
We only want it to conditionally render the action to promote or demote not actually use it as it's value

Reply:
Can we make sure then it gets back only the system mods? I think our auth directive supports it

Reply:
@idoshamun I can only find that option for the whole query, not particularly one field, but maybe I'm missing some knowledge on that.
I also know we can do the nullIfNotLoggedIn kind of thing, but not sure that supports this set.

Reply:
Yes, we can also do that as part of the graphorm if you wish. Because right now the system mods are in-memory it should be easy to do. You can check this function signature, I think you get there the context as well as the query response. Let me know if you need more info